supervisor/options.py

```diff
@@ -2060,6 +2096,51 @@ def make_group(self):
         from supervisor.process import FastCGIProcessGroup
         return FastCGIProcessGroup(self)
 
+
+class DirectoryConfig(object):
+    """Configuration for a required directory."""
+
+    def __init__(self, name, path, create, mode):
+        self.name = name
+        self.path = path
+        self.create = create
+        self.mode = mode
+
+    def __repr__(self):
+        return "DirectoryConfig({!r}, {!r}, {!r}, 0o{:o})".format(
+            self.name, self.path, self.create, self.mode
+        )
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, DirectoryConfig)
+            and self.name == other.name
+            and self.path == other.path
+            and self.create == other.create
+            and self.mode == other.mode
+        )
+
+    def check(self):
+        """Check if the directory exists, and potentially try to
+        create it. Return True if directory was created."""        
+        if os.path.exists(self.path):            
```

Comment:
I see a potential problem here. if a path exists then this if will execute, then the next statement will not (if this will be an actual directory) then the function would exit with False (line 2141). then we go to line 1528 which would raise an error, which is probably not what you want

Reply:
I don't follow. Why would L1528 raise an error? False is returned if no directory was created.

Reply:
ah, you're absolutely right. please ignore this comment